### DISCUSSION TITLE
Why does Memory::view require a store parameter?

### DISCUSSION THREAD

**Discussion by 4LT** [wasmerio/wasmer#3675] [General] [answered]
Why does the present API for the Memory struct require that a store reference be passed to the view method when the Memory is constructed with a store to begin with?  Is it valid to use a different store with this method than the one the Memory was constructed with?

**Comment by jcaesar** [comment, marked as answer]
* Passing a store is required to ensure thread-safe access to the memory. The memory object can be shared and passed around, but the store access is restricted.
* Passing an unrelated store is unlikely to be valid, but I haven't tried.
* (In case you're trying to access the memory from a callback: You can use `Function::new_typed_with_env`, receive a `FunctionEnvMut`, and that implements `AsStoreRef`, giving you access.)
